src/APIBasedAuth.ts

```diff
@@ -133,6 +133,18 @@ class APIBasedAuth {
     return data;
   }
 
+  public async getLoginMethods(username: string) {
+    if (this.loginMethods?.[username]) {
+      return this.loginMethods[username];
+    }
+    const { data } = await this.client.get<
+      APIBasedAuth.LoginMethod[],
+      AxiosResponse<APIBasedAuth.LoginMethod[]>
+    >('/login-methods', { params: { login: username } });
+    this.loginMethods = { [username]: data };
+    return data;
```

Comment:
As-is, I don't think this code allows a client to trigger a fresh request for a username without completely re-instantiating the class -- right?

It seems like most clients would come with their own opinions about caching network results. `ClientUI` does, for example.

Maybe we should skip the local cache here, and just let the client cache results? I feel like most pure API SDKs I've interacted don't seem to do internal caching.